docs/jvb-scaling-management.md

```diff
@@ -0,0 +1,73 @@
+​
+For scalability purposes, it is possible to create a single Jitsi cluster (or shard) with multiple JVB. In this case, each JVB will receive streams of a part of the participants and will send these streams to other participants. Thus each participant will send audio and video stream to a single JVB and receive multiple audio and video streams from one or several JVB, thanks to Octo.
+​
+Therefore, each participant must be able to contact each JVB at any time.
+​
+### Kubernetes context
+As we want to have a scalable infrastructure, Jitsi is deployed on a k8s environment. Each JVB is deployed on a pod which must be accessible from outside the cluster. The number of JVB pod is managed by a Horizontal Pod Autoscaler (HPA).
+​
+This document explains several solutions for the networking issue linked to the variable number of JVB.
+​
+## Additionnal problem
+Kubernetes LoadBalancers does not support mixed loadbalancers (TCP and UDP). (as explained [here](https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/), flags have to be used to make it work).
+Audio and Video streams are sent over UDP. Despite HTTP/TCP, there is few information we can rely on for loadbalancing.
+​
+---
+## Solutions
+---
+### clientIP affinity
+The first solution considered is to create a single service selecting all JVB pods with clientIP affinity. In this case, each user will send and receive streams from a single JVB and will not be able to talk to other JVB.
+​
+In this scenario, octo should be disabled so that a conference is hosted on a single JVB.
+​
+But there is a problem with Network Address Translation (NAT) because as soon as several users are using the same public IP, they can't follow conferences hosted on different JVB.
+​
+A solution could be to write a custom affinity function, but as the routed traffic is built over UDP, the service should look inside the datagram for information.
+​
+Therefore, **this solution is considered invalid**.
+​
+---
+### Service per pod at deployment
+In this solution we have a different service for each JVB pod. The pods are scaling automatically, but the services are all created beforehand. They either link to an existing pod, or to nothing. The pods are deployed thanks to a statefulSet, so that their names follow the logic "jvb-0", "jvb-1"... The services link the jvb-_n_ to port _30300 + n_. The maximum number of jvb pods (defined in the HPA) should be the number of services that are deployed on the cluster.
+​
+In this solution we use a statefulset for JVB pods so that the name of the JVB can be predicted. The goal is to create _n_ services selecting up to _n_ JVB pods, the number of JVB can not be greater than the number of JVB services, this maximum number should be the maximum number of pod in the HPA specification. Every service exposes the underlying pod with a nodePort.
```

Comment:
```suggestion
In this solution we use a statefulset for JVB pods so that the name of the JVB can be predicted. The goal is to create _n_ services selecting up to _n_ JVB pods, the number of JVB can not be greater than the number of JVB services, this maximum number should be the maximum number of pods in the HPA specification. Every service exposes the underlying pod with a nodePort.
```